src/main/scala/org/broadinstitute/dsde/agora/server/dataaccess/permissions/PermissionsClient.scala

```diff
@@ -126,6 +126,25 @@ trait PermissionsClient {
 
   }
 
+  def listOwners(agoraEntity: AgoraEntity): Seq[String] = {
+    val permissionsQuery = for {
+      entity <- entities if entity.alias === alias(agoraEntity)
+      _permissions <- permissions if _permissions.entityID === entity.id // && _permissions.roles===Manage
+      user <- users if user.id === _permissions.userID
+    } yield ( user.email, _permissions.roles)
+
+    val permissionsFuture = db.run(permissionsQuery.result)
+
+    val permissionsRole = permissionsFuture.map { objects: Seq[(String, Int)] =>
+      objects.filter { obj =>
+        AgoraPermissions(obj._2).canManage
+      }.map { obj => obj._1 }
+    } recover {
+      case ex: Throwable => throw new PermissionNotFoundException(s"Could not find owner? ", ex)
+    }
+    Await.result(permissionsRole, timeout)
```

Comment:
Boo. We should avoid `Await.result`, which blocks a thread, and return a Future instead, only map that Future when we really need to.

BUT ... looking at other Agora code, `Await.result` is all over the place, and I'm sure you modeled your code after that. If changing it in this one location is difficult, go ahead and keep it here, and let's file a story to clean it up for real.